src/platform/smart-home-app.ts

```diff
@@ -12,19 +12,24 @@ export class AppStub implements smarthome.App {
   public reachableDevicesHandler:
     | smarthome.IntentFlow.ReachableDevicesHandler
     | undefined;
+  private allHandlersSet: boolean = false;
 
   constructor(version: string) {
     this.version = version;
     //  Allows Local Home Platform to access handlers
     MockLocalHomePlatform.getInstance().setApp(this);
   }
 
+  isAllHandlersSet(): boolean {
+    return this.allHandlersSet;
+  }
+
   getDeviceManager(): smarthome.DeviceManager {
     return MockLocalHomePlatform.getInstance().getDeviceManager();
   }
 
   listen(): Promise<void> {
-    MockLocalHomePlatform.getInstance().notifyHomeAppReady();
+    this.allHandlersSet = true;
```

Comment:
Does it actually guarantee that all the handlers have been set, should we test for it?

Reply:
Added checks and errors for this